plutus-chain-index-core/src/Plutus/ChainIndex/Emulator/Handlers.hs

```diff
@@ -178,20 +179,22 @@ appendBlocks [] = pure ()
 appendBlocks blocks = do
     let
         processBlock (utxoIndexState, txs) (Block tip_ transactions) = do
-            case UtxoState.insert (TxUtxoBalance.fromBlock tip_ (map fst transactions)) utxoIndexState of
-                Left err -> do
-                    let reason = InsertionFailed err
-                    logError $ Err reason
-                    return (utxoIndexState, txs)
-                Right InsertUtxoSuccess{newIndex, insertPosition} -> do
-                    logDebug $ InsertionSuccess tip_ insertPosition
-                    return (newIndex, transactions ++ txs)
+            if null transactions then return (utxoIndexState, txs)
+            else do
```

Comment:
I have looked at `UtxoState.insert` and I suppose there is no point in adding empty `UtxoState`s to the fingertree. We split the fingertree a lot just to add fresh `UtxoState`s without any transactions.

But I would like a validation of the correctness of this.

Reply:
I only see a possible problem with rollbacks.

Reply:
I checked and it should be fine.